qiskit/circuit/controlflow/for_loop.py

```diff
@@ -117,6 +118,12 @@ def params(self, parameters):
     def blocks(self):
         return (self._params[2],)
 
+    def copy_no_body(self):
+        """Return a copy with body arguments replaced by empty circuits with
+        the intention that it will be populated by the caller."""
+        body = QuantumCircuit(self.params[2].qubits, self.params[2].clbits)
+        return ForLoopOp(copy(self.params[0]), copy(self.params[1]), body, self.label)
```

Comment:
Both the index set and the parameter are immutable (`tuple[int]`/`range` and `Parameter`/`None` respectively), so no need to copy them.